internal/engine/experiment/signal/signal.go

```diff
@@ -0,0 +1,177 @@
+IPPQLocLn8GWLwOYNnYfBvILKDMItmZTtEbucdigxEA9mfIvvHADEbteLtVgwBm9
+R5vVvtwrD6CCxI3pgH7EH7kMP0Od93wLisvn1yhHY7FuYlrkYqdkMvWUrKoASVw4
+jb69vaeJCUdU+HCoXOSP1PQcL6WenNCHAgMBAAGjUDBOMB0GA1UdDgQWBBQBixjx
+P/s5GURuhYa+lGUypzI8kDAfBgNVHSMEGDAWgBQBixjxP/s5GURuhYa+lGUypzI8
+kDAMBgNVHRMEBTADAQH/MA0GCSqGSIb3DQEBBQUAA4IBAQB+Hr4hC56m0LvJAu1R
+K6NuPDbTMEN7/jMojFHxH4P3XPFfupjR+bkDq0pPOU6JjIxnrD1XD/EVmTTaTVY5
+iOheyv7UzJOefb2pLOc9qsuvI4fnaESh9bhzln+LXxtCrRPGhkxA1IMIo3J/s2WF
+/KVYZyciu6b4ubJ91XPAuBNZwImug7/srWvbpk0hq6A6z140WTVSKtJG7EP41kJe
+/oF4usY5J7LPkxK3LWzMJnb5EIJDmRvyH8pyRwWg6Qm6qiGFaI4nL8QU4La1x2en
+4DGXRaLMPRwjELNgQPodR38zoCMuA8gHZfZYYoZ7D7Q1wNUiVHcxuFrEeBaYJbLE
+rwLV
+-----END CERTIFICATE-----`
+)
+
+// Config contains the signal experiment config.
+type Config struct{}
+
+// TestKeys contains signal test keys.
+type TestKeys struct {
+	urlgetter.TestKeys
+	SignalBackendStatus  string  `json:"signal_backend_status"`
+	SignalBackendFailure *string `json:"signal_backend_failure"`
+}
+
+// NewTestKeys creates new signal TestKeys.
+func NewTestKeys() *TestKeys {
+	return &TestKeys{
+		SignalBackendStatus:  "ok",
+		SignalBackendFailure: nil,
+	}
+}
+
+// Update updates the TestKeys using the given MultiOutput result.
+func (tk *TestKeys) Update(v urlgetter.MultiOutput) {
+	// update the easy to update entries first
+	tk.NetworkEvents = append(tk.NetworkEvents, v.TestKeys.NetworkEvents...)
+	tk.Queries = append(tk.Queries, v.TestKeys.Queries...)
+	tk.Requests = append(tk.Requests, v.TestKeys.Requests...)
+	tk.TCPConnect = append(tk.TCPConnect, v.TestKeys.TCPConnect...)
+	tk.TLSHandshakes = append(tk.TLSHandshakes, v.TestKeys.TLSHandshakes...)
+	if v.TestKeys.Failure != nil {
+		tk.SignalBackendStatus = "blocked"
+		tk.SignalBackendFailure = v.TestKeys.Failure
+		return
+	}
+	return
+}
+
+// Measurer performs the measurement
+type Measurer struct {
+	// Config contains the experiment settings. If empty we
+	// will be using default settings.
+	Config Config
+
+	// Getter is an optional getter to be used for testing.
+	Getter urlgetter.MultiGetter
+}
+
+// ExperimentName implements ExperimentMeasurer.ExperimentName
+func (m Measurer) ExperimentName() string {
+	return testName
+}
+
+// ExperimentVersion implements ExperimentMeasurer.ExperimentVersion
+func (m Measurer) ExperimentVersion() string {
+	return testVersion
+}
+
+// Run implements ExperimentMeasurer.Run
+func (m Measurer) Run(ctx context.Context, sess model.ExperimentSession,
+	measurement *model.Measurement, callbacks model.ExperimentCallbacks) error {
+	ctx, cancel := context.WithTimeout(ctx, 60*time.Second)
+	defer cancel()
+	urlgetter.RegisterExtensions(measurement)
+
+	certPool := netx.NewDefaultCertPool()
+	if certPool.AppendCertsFromPEM([]byte(signalCA)) == false {
+		return errors.New("AppendCertsFromPEM failed")
+	}
+
+	inputs := []urlgetter.MultiInput{
+		// Here we need to provide the method explicitly. See
+		// https://github.com/ooni/probe-cli/v3/internal/engine/issues/827.
```

Comment:
LOL, this is a URL that should not have changed when I integrated probe-cli into probe-engine :facepalm: :grimacing: